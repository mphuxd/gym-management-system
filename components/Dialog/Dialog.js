import React from 'react';
import * as DialogPrimitive from '@radix-ui/react-dialog';
import cx from 'classnames';
import { DialogOverlay } from '@/components';

const Dialog = React.forwardRef(
  ({ children, open, onOpenChange, className, ...props }, forwardedRef) => (
    <DialogPrimitive.Root
      open={open}
      onOpenChange={onOpenChange}
      ref={forwardedRef}
      className={cx(className, '')}
      {...props}
    >
      <DialogPrimitive.Portal className="relative z-20">
        <DialogOverlay className="z-10" />
        {children}
      </DialogPrimitive.Portal>
    </DialogPrimitive.Root>
  )
);

Dialog.displayName = 'Dialog';

export default Dialog;
