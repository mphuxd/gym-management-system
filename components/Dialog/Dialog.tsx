import React from 'react';
import * as DialogPrimitive from '@radix-ui/react-dialog';
import { DialogOverlay, DialogTrigger } from '@/components';

export interface DialogProps extends DialogPrimitive.DialogProps {
  trigger: string;
}

function Dialog({
  children,
  open,
  onOpenChange,
  trigger,
  ...props
}: DialogProps) {
  <DialogPrimitive.Root open={open} onOpenChange={onOpenChange} {...props}>
    {trigger && <DialogTrigger>{trigger}</DialogTrigger>}
    <DialogPrimitive.Portal className="relative z-20">
      <DialogOverlay />
      {children}
    </DialogPrimitive.Portal>
  </DialogPrimitive.Root>;
}

Dialog.displayName = 'Dialog';

export default Dialog;
