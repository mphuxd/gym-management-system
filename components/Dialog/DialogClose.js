import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';
import { Cross2Icon } from '@radix-ui/react-icons';

const DialogClose = React.forwardRef(
  ({ className, ...props }, forwardedRef) => (
    <DialogPrimitive.Close
      {...props}
      ref={forwardedRef}
      className={cx(
        className,
        'bg-transparent p-2 hover:bg-neg-hover hover:text-white active:bg-neg-active active:outline-none'
      )}
    >
      <Cross2Icon className="h-4 w-4 fill-icon" />
    </DialogPrimitive.Close>
  )
);

DialogClose.displayName = 'DialogClose';

export default DialogClose;
