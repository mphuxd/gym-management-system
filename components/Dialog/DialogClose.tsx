import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';
import { Cross2Icon } from '@radix-ui/react-icons';

type DialogCloseRef = React.ElementRef<typeof DialogPrimitive.Close>;
export interface DialogCloseProps extends DialogPrimitive.DialogCloseProps {}

const DialogClose = React.forwardRef<
  DialogCloseRef,
  DialogPrimitive.DialogCloseProps
>(({ className, ...props }, forwardedRef) => (
  <DialogPrimitive.Close
    ref={forwardedRef}
    className={cx(
      className,
      'bg-transparent p-2 hover:bg-neg-hover hover:text-white active:bg-neg-active active:outline-none'
    )}
    {...props}
  >
    <Cross2Icon className="h-4 w-4 fill-icon" />
  </DialogPrimitive.Close>
));

DialogClose.displayName = 'DialogClose';

export default DialogClose;
