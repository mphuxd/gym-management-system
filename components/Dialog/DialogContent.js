import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';

const DialogContent = React.forwardRef(
  ({ children, className, title, ...props }, forwardedRef) => {
    const classNames = cx(className, 'absolute bg-white rounded-lg z-50');
    return (
      <DialogPrimitive.Content
        {...props}
        ref={forwardedRef}
        className={classNames}
      >
        {children}
      </DialogPrimitive.Content>
    );
  }
);

DialogContent.displayName = 'DialogContent';

export default DialogContent;
