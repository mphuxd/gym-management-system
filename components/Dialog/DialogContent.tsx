import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';

const DialogContent = React.forwardRef(
  ({ children, className, title, rounded = true, ...props }, forwardedRef) => {
    const classNames = cx(className, 'absolute z-30 bg-white', {
      'rounded-lg': rounded,
    });
    return (
      <DialogPrimitive.Content
        {...props}
        ref={forwardedRef}
        className={classNames}
      >
        {children}
      </DialogPrimitive.Content>
    );
  }
);

DialogContent.displayName = 'DialogContent';

export default DialogContent;
