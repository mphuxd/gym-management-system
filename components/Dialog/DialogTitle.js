import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';

const DialogTitle = React.forwardRef(
  ({ children, className, ...props }, forwardedRef) => (
    <DialogPrimitive.Title
      {...props}
      ref={forwardedRef}
      className={cx(className, 'text-lg font-bold')}
    >
      {children}
    </DialogPrimitive.Title>
  )
);

DialogTitle.displayName = 'DialogTitle';

export default DialogTitle;
