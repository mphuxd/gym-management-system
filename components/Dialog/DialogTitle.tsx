import React from 'react';
import cx from 'classnames';
import * as DialogPrimitive from '@radix-ui/react-dialog';

type DialogTitleRef = React.ElementRef<typeof DialogPrimitive.Content>;
export interface DialogTitleProps extends DialogPrimitive.DialogTitleProps {}

const DialogTitle = React.forwardRef<DialogTitleRef, DialogTitleProps>(
  ({ children, className, ...props }, forwardedRef) => (
    <DialogPrimitive.Title
      className={cx(className, 'text-lg font-bold')}
      ref={forwardedRef}
      {...props}
    >
      {children}
    </DialogPrimitive.Title>
  )
);

DialogTitle.displayName = 'DialogTitle';

export default DialogTitle;
