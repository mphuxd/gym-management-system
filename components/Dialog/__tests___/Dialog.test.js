import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Dialog from '../Dialog';
import DialogContent from '../DialogContent';
import DialogClose from '../DialogClose';
import DialogTitle from '../DialogTitle';

describe('Dialog: ', () => {
  it('should match snapshot', async () => {
    render(
      <div>
        <Dialog trigger="Trigger">
          <DialogContent>
            <DialogTitle>Test Dialog</DialogTitle>
            <DialogClose />
          </DialogContent>
        </Dialog>
      </div>
    );

    const trigger = await screen.findByText('Trigger');
    await userEvent.click(trigger);
    const dialog = await screen.findByRole('dialog');
    expect(dialog).toMatchSnapshot();
  });
});
