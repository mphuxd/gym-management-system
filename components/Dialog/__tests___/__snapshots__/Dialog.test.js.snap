// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog:  should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="absolute bg-white rounded-lg z-50"
  data-state="open"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <h2
    class="font-bold text-lg"
    id="radix-:r1:"
  >
    Test Dialog
  </h2>
  <button
    class="p-2 block hover:bg-slate4 rounded-sm focus:outline focus:outline-slate7 active:bg-slate5"
    type="button"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 15 15"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
